tests/test_fcp.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from unittest import mock
+
+import pytest
+
+from constructor.fcp import check_duplicates
+
+
+class GenericObject:
+    """We use this for testing the check_duplicates function"""
+    def __init__(self, name):
+        self.name = name
+        self.fn = 'filename.txt'
+
+
+@pytest.mark.parametrize('values,expected_fails', (
+    (
+        (GenericObject('NameOne'), GenericObject('NameTwo'), GenericObject('NameTwo')), 1
+    ),
+    (
+        (GenericObject('NameOne'), GenericObject('NameTwo')), 0
+    ),
+    (
+        (GenericObject('NameOne'), GenericObject('NameTwo'), GenericObject('NameThree')), 0
+    ),
+))
+def test_check_duplicates(values: tuple[..., GenericObject], expected_fails: int):
+    with mock.patch('constructor.fcp.sys.exit') as sys_exit:
+        check_duplicates(values)
+        assert len(sys_exit.mock_calls) == expected_fails
```

Comment:
`sys.exit` raises `SystemExit`, no need to mock, see https://docs.python.org/3.10/library/sys.html#sys.exit

```python
from contextlib import nullcontext

class _NameFn:
    """We use this for testing the check_duplicates function"""
    def __init__(self, name):
        self.name = name
        self.fn = f'filename/{name}.txt'


@pytest.mark.parametrize('values,expected_fails', (
    ((_NameFn('One'), _NameFn('Two'), _NameFn('Two')), True),
    ((_NameFn('One'), _NameFn('Two')), False),
    ((_NameFn('One'), _NameFn('Two'), _NameFn('Three')), False),
))
def test_check_duplicates_failures(values: tuple[_NameFn, ...], expected_fails: bool):
    if expected_fails:
        context = pytest.raises(SystemExit)
    else:
        context = nullcontext()

    with context:
        check_duplicates(values)
```

Reply:
Cool use of `nullcontext`. I will update the tests.